fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/ResourceFactoryImpl.java

```diff
@@ -100,6 +100,12 @@ public WebacAcl createAcl(final Transaction transaction, final String identifier
         return (WebacAcl) createResource(WebacAcl.class);
     }
 
+    @Override
+    public FedoraResource getResource(final String identifier)
+            throws PathNotFoundException {
+        return getResource(null, identifier);
```

Comment:
I think it's cleaner for getResource(transaction, identifier) to throw an exception on null. Both methods will call into the PersistentStorageSession which also has separate methods for requesting transactional vs immutable state.

Reply:
above should be a change request:

Reply:
We don't specify any exceptions being thrown in either the ResourceFactory or PersistentStorageManager.

The exception you are referencing is thrown by the OCFL implementation. The ResourceFactoryImpl is not aware of that specific implementation.

But regardless, this (ResourceFactory) is a separate API. It (IMHO) does not have to mimic the implementation that it is under it.

Reply:
The exception I was thinking was a runtime IllegalArgumentException. 

I'm okay skipping the overloaded method since I don't believe we need support for it at this time (though we likely we need it later when we implement memento support).

Reply:
The question I have is why throw an Exception from the Persistence layer all the way to the HTTP layer? Wouldn't it make more sense to check that `Atomic-Id != null` at the top? 

Instead it seems to me (correct me if I'm wrong) you are proposing adding code to keep a clear path from HTTP -> Persistence to ensure the `null` gets passed down and then fails.

If an `Atomic-Id` derived from the header does not refer to a valid transaction, then you'd want to throw an Exception but again that should occur before you try to use the `Atomic-Id`. I'd say in some HTTP layer utility like [here](https://github.com/fcrepo4/fcrepo4/pull/1560/files#diff-3f238be750488e3730daa409a69174b6R116)

Reply:
I think we're on the same page @whikloj.  I wasn't suggesting we throw an exception due to a missing transaction id in the persistence layer.